src/databricks/labs/ucx/source_code/base.py

```diff
@@ -1,15 +1,20 @@
 from __future__ import annotations
 
+import sys
+
 from abc import abstractmethod
 from collections.abc import Iterable
 from dataclasses import dataclass
 from pathlib import Path
 
 from astroid import NodeNG  # type: ignore
-from typing_extensions import Self
 
 from databricks.sdk.service import compute
 
+if sys.version_info[1] < 11:
```

Comment:
The idiomatic check for this is:
```suggestion
if sys.version_info >= (3, 11):
```